src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  const groups: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    if (groups[item[key]]) {
+      groups[item[key]].push(item);
+    } else {
+      groups[item[key]] = [item];
+    }
```

Comment:
```suggestion
    if (!groups[item[key]]) {
      groups[item[key]] = [];
    }
    groups[item[key]].push(item);
```

And here you can avoid else statement if you will check your key on false for the first.
There can be problem with a type of `item[key]`, so you should convert it to string.